cmake/tests/cxx14_auto.cpp

```diff
@@ -0,0 +1,20 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2009 Andrey Semashev
+//  Copyright (c) 2015 Agustin Berge
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+template<typename T>
+auto check_f(T& i)
+{
+    T j = 1 + j;
+    return j;
+}
```

Comment:
Let me rephrase this. If the feature test is for a C++14 feature (which is fine), why do you use the result of this feature test to add a workaround for a missing C++11 capability of a particular compiler? Shouldn't we rather feature test for the particular missing capabilities und use this instead to decide whether to activate the workaround?